bacs/src/main/java/com/adyen/checkout/bacs/DefaultBacsDirectDebitDelegate.kt

```diff
@@ -39,11 +75,24 @@ internal class DefaultBacsDirectDebitDelegate(
             mode = inputData.mode,
         )
 
+        updateViewType(inputData.mode)
+
         _outputDataFlow.tryEmit(outputData)
 
         createComponentState(outputData)
     }
 
+    private fun updateViewType(mode: BacsDirectDebitMode) {
+        val viewType = when (mode) {
+            BacsDirectDebitMode.INPUT -> BacsComponentViewType.INPUT
+            BacsDirectDebitMode.CONFIRMATION -> BacsComponentViewType.CONFIRMATION
+        }
+        if (_viewFlow.value != viewType) {
```

Comment:
This check is not really needed, because `StateFlow`s basically have a built in `distinctUntilChanged`

Reply:
Hmm yeah but I'd still like to log a line only when it's different